src/hooks/use-document-sync-to-firebase.ts

```diff
@@ -78,6 +78,48 @@ export function useDocumentSyncToFirebase(
     }
   });
 
+  // OPTION 1 of 3 sync visibility (public/private) for personal documents to typedMetadata
+  useSyncMstPropToFirebase<typeof document.visibility>({
+    firebase, model: document, prop: "visibility", path: typedMetadata,
+    enabled: commonSyncEnabled && !readOnly && [PersonalDocument, LearningLogDocument].includes(type),
+    options: {
+      onSuccess: (data, visibility) => {
+        debugLog(`DEBUG: Updated document visibility for ${type} document ${key}:`, visibility);
+      },
+      onError: (err, visibility) => {
+        console.warn(`ERROR: Failed to update document visibility for ${type} document ${key}:`, visibility);
+      }
+    }
+  });
+
+  // OPTION 2 of 3 sync visibility (public/private) for personal documents to metadata
+  useSyncMstPropToFirebase<typeof document.visibility>({
```

Comment:
`useSyncMstPropToFirebase` will only sync the data from local to firebase. 
So in order for this data to be loaded in again. You'll probably have to update the code goes the other way (sync from firebase to local). This is the code which is loading the documents and also watching the meta data for changes. I'd suspect this code is not looking for a visibility property in the metadata and the setting it as 'document.visbility'.